### PR TITLE
Roc MB-3173 counselor add shipment

### PR DESCRIPTION
**Pull request by Ronolibert**
## Description

Add a create MTO Shipment endpoint to the GHC API and update prime/ghc payloads to include counselor remark field

## Reviewer notes
* Consistent with prime API createMTOShipment endpoint but simplified functionality
* Prime API MTOShipment payload also updated to have CounselorRemark field

## Setup
* [Set up postman](https://github.com/transcom/mymove/wiki/Test-Admin-and-Office-APIs-with-Postman)
* Make POST request to /mto-shipments with body:
```
{
    "moveTaskOrderID": "<movetaskorder uuid>",
    "requestedPickupDate": "2021-07-16",
    "pickupAddress": {
        "street_address_1": "culpa do ut cupidatat",
        "city": "adipisicing proident quis",
        "state": "CT",
        "postal_code": "11750-5518",
        "street_address_2": "nulla anim est enim",
        "street_address_3": "fugiat eiusmod consectetur dolor",
        "eTag": "tempor magna dolore veniam",
        "country": "USA"
    },
    "destinationAddress": {
        "street_address_1": "aute exercitation ad id dolor",
        "city": "id",
        "state": "MN",
        "postal_code": "60660",
        "street_address_2": "adipisicing ipsum occ",
        "street_address_3": "tempor adipisicing aliquip anim Duis",
        "eTag": "sunt",
        "country": "USA"
    },
    "customerRemarks": "customer remarks",
    "counselorRemarks": "counselor remarks",
    "agents": [
        {
            "createdAt": "2011-02-25T10:58:16.685Z",
            "updatedAt": "1977-12-04T09:18:35.908Z",
            "firstName": "ut enim",
            "lastName": "Ut Duis ullamco",
            "email": "a-wZZ2%1@1ofbv4g.kBKdnHiRqwXF",
            "phone": "978-787-5714",
            "agentType": "RELEASING_AGENT",
            "eTag": "in ex"
        },
        {
            "createdAt": "2019-03-19T06:33:15.864Z",
            "updatedAt": "1961-02-07T06:20:34.151Z",
            "firstName": "proident incididunt de",
            "lastName": "proident dolor aute amet id",
            "email": "-x3SVygu@A9zyh.LxrOiBwy",
            "phone": "863-853-9664",
            "agentType": "RECEIVING_AGENT",
            "eTag": "pariatur"
        }
    ]
}
```
Expected result: MTO Shipment should be created in the DB

## Code Review Verification Steps
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3173) for this change

### CODE REVIEW

**Review by lynzt — COMMENTED** [review]
LGTM - can you add a test to verify the counselor remarks on the primeapi?

**Review by sandy-wright — CHANGES_REQUESTED** [review]
LGTM! But there is a potential bug in the `AddressModel` function (my bad 😅 ) that we should go ahead and fix before merging with master. It's fine for now, but it'll become a problem if you add an update endpoint later on.

File: pkg/handlers/ghcapi/mto_shipment.go [hunk 1]
```diff
@@ -82,6 +82,49 @@ func (h ListMTOShipmentsHandler) Handle(params mtoshipmentops.ListMTOShipmentsPa
 	return mtoshipmentops.NewListMTOShipmentsOK().WithPayload(*payload)
 }
 
+// CreateMTOShipmentHandler is the handler to create MTO shipments
+type CreateMTOShipmentHandler struct {
+	handlers.HandlerContext
+	mtoShipmentCreator     services.MTOShipmentCreator
+	mtoAvailabilityChecker services.MoveTaskOrderChecker
```
[the remark this answers is not in the document]
Comment (Ronolibert) [reply pkg/handlers/ghcapi/mto_shipment.go:89]: Woops! I do not need it in the GHC API, will remove this

File: pkg/handlers/ghcapi/mto_shipment.go [hunk 2]
```diff
@@ -82,6 +82,49 @@ func (h ListMTOShipmentsHandler) Handle(params mtoshipmentops.ListMTOShipmentsPa
...
+	}
+
+	mtoShipment := payloads.MTOShipmentModelFromCreate(payload)
+	mtoShipment, err := h.mtoShipmentCreator.CreateMTOShipment(mtoShipment, nil)
+
+	if err != nil {
+		logger.Error("ghcapi.CreateMTOShipmentHandler error", zap.Error(err))
+		switch e := err.(type) {
+		case services.NotFoundError:
+			return mtoshipmentops.NewCreateMTOShipmentNotFound()
```
[the remark this answers is not in the document]
Comment (Ronolibert) [reply pkg/handlers/ghcapi/mto_shipment.go:109]: Looks like there needs to be some standardization on how errors are handled in the GHC side. In the prime, we use clientErrors with title, detail and instance but on GHC our 404s are just Errors with a message field in them. I'm just gonna use a message for now

[the remark this answers is not in the document]
Comment (sandy-wright) [reply pkg/handlers/ghcapi/mto_shipment.go]: Ohhh okay. That's fair!

File: pkg/handlers/ghcapi/mto_shipment_test.go [hunk 3]
```diff
@@ -449,3 +453,172 @@ func (suite *HandlerSuite) TestPatchMTOShipmentHandler() {
...
+		}
+		response := handler.Handle(params)
+		okResponse := response.(*mtoshipmentops.CreateMTOShipmentOK)
+		createMTOShipmentPayload := okResponse.Payload
+		suite.IsType(&mtoshipmentops.CreateMTOShipmentOK{}, response)
+
+		suite.Require().Equal(createMTOShipmentPayload.Status, ghcmessages.MTOShipmentStatusSUBMITTED, "MTO Shipment should have been submitted")
+		suite.Require().Equal(createMTOShipmentPayload.ShipmentType, ghcmessages.MTOShipmentTypeHHG, "MTO Shipment should be an HHG")
+		suite.Equal(*createMTOShipmentPayload.CustomerRemarks, string("customer remark"))
+		suite.Equal(*createMTOShipmentPayload.CounselorRemarks, string("counselor remark"))
```
[the remark this answers is not in the document]
Comment (Ronolibert) [reply pkg/handlers/ghcapi/mto_shipment_test.go]: I saw this being used in the prime API! I think there is some short circuit behavior with require, but totally just a nice to have

**Review by sandy-wright — APPROVED** [review]
LGTM! Thanks for making those changes! 🎉

**Pull request merged**